### PR TITLE
[SPARK-44923][PYTHON][BUILD] Some directories should be cleared when regenerating files

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to fix some bug in regenerating pyspark docs in certain scenarios.

### Why are the changes needed?
- The following error occurred while I was regenerating the pyspark document.
   image

- We can simply reproduce this problem as follows:
 1.git reset --hard 3f380b9ecc8b27f6965b554061572e0990f0513
    image
 2.make clean html, (at this point, it is successful.)
    image
3.git pull, (at this point, the function `chr` has been deleted, but the previously generated file(`pyspark.sql.functions.chr.rst`) will not be deleted.)
4.make clean html, (at this point, it is failed.)
    image

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
1.Pass GA.
2.Manually test.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
As shown in the following figure, during the document generation process of `sphinx-build`, some directories and files will be automatically generated in the directory: 
image
image

**Comment by panbingkun** [comment]
cc @zhengruifeng @HyukjinKwon

**Comment by zhengruifeng** [comment]
@panbingkun in what case will this be a problem?
I don't see similar doc build failures in CI of master and branch-3.5

**Comment by panbingkun** [comment]
> in what case will this be a problem?
> I don't see similar doc build failures in CI of master and branch-3.5

1. The docs file has been generated locally.
2. At this point, the pyspark code has made changes, such as deleting the function 'chr'`.
3. Execute the command to generate docs files locally, and an error will occur at this time.
4. 
The reason why it doesn't happen on GA is because it's always a new generation on GA.

In `conf.py`, both directories `source\reference\api` and `reference\pyspark.pandas\api` have been cleaned, and we should maintain consistency: `reference\pyspark.sql\api` and `reference\pyspark.ss\api`.

**Review by zhengruifeng — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

File: python/docs/source/conf.py [hunk 1]
```diff
@@ -33,22 +33,16 @@
...
-except OSError as e:
-    if e.errno != errno.EEXIST:
-        raise
-try:
-    os.mkdir("%s/reference/pyspark.pandas/api" % os.path.dirname(
-        os.path.abspath(__file__)))
-except OSError as e:
-    if e.errno != errno.EEXIST:
-        raise
+gen_rst_dirs = ["reference/api", "reference/pyspark.pandas/api",
```
Comment (panbingkun) [inline python/docs/source/conf.py:36]: Make the code more concise.

**Comment by panbingkun** [comment]
Compared to the 1st version, the 2nd version makes the code more concise.

**Pull request closed without merging**